rules/anti-static/obfuscation/js.yara

```diff
@@ -3,9 +3,14 @@ import "math"
 private rule probably_js {
   strings:
     $f_function = /function\(\w{0,8}\)/
-    $f_const    = "const" fullword
-    $f_return   = "return" fullword
-    $f_var      = "var" fullword
+    $f_const    = /\bconst\b/
```

Comment:
Isn't `"\b<text>\b"` and `"<text>" fullword` equivalent?

Reply:
`fullword` will match non-alphanumeric delimited strings, so things like `my-const` or `a-var` or `_return` will also match IIRC.

Reply:
I did some testing, and as best as I can tell, `fullword` is equivalent to `\bword\b`. Tested with this rule:

```
rule const {
  strings:
    $fullword = "const" fullword
    $const_b_both = /\bconst\b/
    $const_s_both = /\sconst\s/
    $const_maybe_s_both = /\s{0,18}const\s/
    $const_anchor_space = /^\s{0,18}const\s/
  condition:
    any of them
}
```

```
# paste rule in
cat > /tmp/const.yara 

echo "const PI = 3.141569" >/tmp/const.js                          echo "/etc/const/" > /tmp/const.path
yara -s -w /tmp/const.yara /tmp/const.js
yara -s -w /tmp/const.yara /tmp/const.path
```

The strictest regex that works for this example is 
 `const_anchor_space`, followed by `const_maybe_s_both`.

When run against `/tmp/const.path`, the following match inadvertently:

```
0x5:$fullword: const
0x5:$const_b_both: const
```